iam/api-client/src/main/java/iam/snippets/CreateServiceAccountKey.java

```diff
@@ -16,80 +16,36 @@
-import com.google.api.services.iam.v1.model.CreateServiceAccountKeyRequest;
-import com.google.api.services.iam.v1.model.ServiceAccountKey;
-import com.google.auth.http.HttpCredentialsAdapter;
-import com.google.auth.oauth2.GoogleCredentials;
+
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.iam.admin.v1.CreateServiceAccountKeyRequest;
+import com.google.iam.admin.v1.ServiceAccountKey;
 import java.io.IOException;
-import java.security.GeneralSecurityException;
-import java.util.Base64;
-import java.util.Collections;
 
 public class CreateServiceAccountKey {
 
   // Creates a key for a service account.
-  public static String createKey(String projectId, String serviceAccountName) {
-    // String projectId = "my-project-id";
-    // String serviceAccountName = "my-service-account-name";
-
-    Iam service = null;
-    try {
-      service = initService();
-    } catch (IOException | GeneralSecurityException e) {
-      System.out.println("Unable to initialize service: \n" + e);
-      return null;
-    }
-
-    // Construct the service account email.
-    // You can modify the ".iam.gserviceaccount.com" to match the service account name in which
-    // you want to create the key.
-    // See, https://cloud.google.com/iam/docs/creating-managing-service-account-keys?hl=en#creating
-    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
-    try {
-      ServiceAccountKey key =
-          service
-              .projects()
-              .serviceAccounts()
-              .keys()
-              .create(
-                  "projects/-/serviceAccounts/" + serviceAccountEmail,
-                  new CreateServiceAccountKeyRequest())
-              .execute();
-
-      // The privateKeyData field contains the base64-encoded service account key
-      // in JSON format.
-      // TODO(Developer): Save the below key (jsonKeyFile) to a secure location.
-      //  You cannot download it later.
-      String jsonKeyFile = new String(Base64.getDecoder().decode(key.getPrivateKeyData()));
-
+  public static String createKey(String projectId, String accountName) {
+    // String projectId = "project-id";
+    // String accountName = "my-service-account-name";
+    String email = accountName + "@" + projectId + ".iam.gserviceaccount.com";
+    try (IAMClient iamClient = IAMClient.create()) {
+      CreateServiceAccountKeyRequest req = createAccountKeyReq(projectId, email);
+      ServiceAccountKey createdKey = iamClient.createServiceAccountKey(req);
       System.out.println("Key created successfully");
-      String keyName = key.getName();
+
+      String keyName = createdKey.getName();
       return keyName.substring(keyName.lastIndexOf("/") + 1).trim();
-    } catch (IOException e) {
-      System.out.println("Unable to create service account key: \n" + e);
+    } catch (IOException ex) {
+      System.out.println("Failed to create key");
       return null;
```

Comment:
Do not catch the exception unless it's intended to show how to handle the exception. If it's a print statement, avoid catching the exception and instead throw them. (Applies to all samples)